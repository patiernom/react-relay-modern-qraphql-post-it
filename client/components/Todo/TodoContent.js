import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';

const TodoContent = ({ text, timestamp }) => (
  <div>
    <div className={'content-text'}>{text}</div>
    <span className={'time'}>
      {moment(timestamp).format('MMMM Do YYYY, h:mm:ss a')}
    </span>
  </div>
);

TodoContent.propTypes = {
  text: PropTypes.string.isRequired,
  timestamp: PropTypes.node.isRequired
};

TodoContent.defaultProps = {};

export default TodoContent;