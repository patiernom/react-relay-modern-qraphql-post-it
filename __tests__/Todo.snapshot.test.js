import React from 'react';
import renderer from 'react-test-renderer';
import TodoContent from '../client/components/Todo/TodoContent';

const dateTime = 1514309900222; // December 26th 2017, 6:38:20 pm

describe('Todo component renders the todo correctly', () => {
  it('renders correctly', () => {
    const todo = {text: 'Buy Milk', timestamp: dateTime };

    const rendered = renderer.create(<TodoContent text={todo.text} timestamp={todo.timestamp} />);

    expect(rendered.toJSON()).toMatchSnapshot();
  });
});