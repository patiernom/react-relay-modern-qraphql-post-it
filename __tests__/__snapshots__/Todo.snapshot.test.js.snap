// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component renders the todo correctly renders correctly 1`] = `
<div>
  <div
    className="content-text"
  >
    Buy Milk
  </div>
  <span
    className="time"
  >
    December 26th 2017, 6:38:20 pm
  </span>
</div>
`;
